### PR TITLE
Add an option to force public replays for room tournaments

### PR DESCRIPTION
**Pull request by AustinXII**

### CODE REVIEW

File: server/tournaments/index.ts [hunk 1]
```diff
@@ -1524,6 +1526,22 @@ const tourCommands: {basic: TourCommands, creation: TourCommands, moderation: To
 				return this.sendReply(`Usage: ${cmd} <allow|disallow>`);
 			}
 		},
+		forcepublicreplays(tournament, user, params, cmd) {
+			const option = params.length ? params[0].toLowerCase() : 'on';
+			if (this.meansYes(option)) {
+				tournament.forcePublicReplays = true;
+				this.room.add('forcepublicreplays is now on for the tournament.');
+				this.privateModAction(`(Forced public replays was turned on for the tournament by ${user.name})`);
+				this.modlog('TOUR fORCEPUBLICREPLAYS', null, 'ON');
```
Comment (Zarel) [inline server/tournaments/index.ts:1535]:
```suggestion
				this.modlog('TOUR FORCEPUBLICREPLAYS', null, 'ON');
```

File: server/tournaments/index.ts [hunk 2]
```diff
@@ -1524,6 +1526,22 @@ const tourCommands: {basic: TourCommands, creation: TourCommands, moderation: To
 				return this.sendReply(`Usage: ${cmd} <allow|disallow>`);
 			}
 		},
+		forcepublicreplays(tournament, user, params, cmd) {
+			const option = params.length ? params[0].toLowerCase() : 'on';
+			if (this.meansYes(option)) {
+				tournament.forcePublicReplays = true;
+				this.room.add('forcepublicreplays is now on for the tournament.');
```
Comment (Zarel) [inline server/tournaments/index.ts:1533]:
```suggestion
				this.room.add('Tournament replays forced public: ON');
```

File: server/tournaments/index.ts [hunk 3]
```diff
@@ -1524,6 +1526,22 @@ const tourCommands: {basic: TourCommands, creation: TourCommands, moderation: To
...
+				this.privateModAction(`(Forced public replays was turned on for the tournament by ${user.name})`);
+				this.modlog('TOUR fORCEPUBLICREPLAYS', null, 'ON');
+			} else if (this.meansNo(option) || option === 'stop') {
+				tournament.forcePublicReplays = false;
+				this.room.add('Forcepublicreplays is now off for the tournament.');
```
Comment (Zarel) [inline server/tournaments/index.ts:1538]:
```suggestion
				this.room.add('Tournament replays forced public: OFF');
```

File: server/tournaments/index.ts [hunk 4]
```diff
@@ -1524,6 +1526,22 @@ const tourCommands: {basic: TourCommands, creation: TourCommands, moderation: To
 				return this.sendReply(`Usage: ${cmd} <allow|disallow>`);
 			}
 		},
+		forcepublicreplays(tournament, user, params, cmd) {
+			const option = params.length ? params[0].toLowerCase() : 'on';
+			if (this.meansYes(option)) {
+				tournament.forcePublicReplays = true;
+				this.room.add('forcepublicreplays is now on for the tournament.');
+				this.privateModAction(`(Forced public replays was turned on for the tournament by ${user.name})`);
```
Comment (Zarel) [inline server/tournaments/index.ts:1534]:
```suggestion
				this.privateModAction(`(Tournament public replays were turned ON by ${user.name})`);
```

File: server/tournaments/index.ts [hunk 5]
```diff
@@ -1524,6 +1526,22 @@ const tourCommands: {basic: TourCommands, creation: TourCommands, moderation: To
...
+				this.modlog('TOUR fORCEPUBLICREPLAYS', null, 'ON');
+			} else if (this.meansNo(option) || option === 'stop') {
+				tournament.forcePublicReplays = false;
+				this.room.add('Forcepublicreplays is now off for the tournament.');
+				this.privateModAction(`(Forced public replays was turned off for the tournament by ${user.name})`);
```
Comment (Zarel) [inline server/tournaments/index.ts:1539]:
```suggestion
				this.privateModAction(`(Tournament public replays were turned on by ${user.name})`);
```

**Comment by Zarel** [comment]
Good change!

**Comment by AustinXII** [comment]
@Zarel done

File: server/tournaments/index.ts [hunk 6]
```diff
@@ -1524,6 +1526,22 @@ const tourCommands: {basic: TourCommands, creation: TourCommands, moderation: To
 				return this.sendReply(`Usage: ${cmd} <allow|disallow>`);
 			}
 		},
+		forcepublicreplays(tournament, user, params, cmd) {
+			const option = params.length ? params[0].toLowerCase() : 'on';
```
Comment (Zarel) [inline server/tournaments/index.ts:1530]:
```suggestion
			const option = params[0] || 'on';
```

`meansYes` and `meansNo` already normalize this.

**Comment by AustinXII** [comment]
@Zarel done

**Comment by Zarel** [comment]
I don't see anything that forces the games to be public, though. Using `/hiddenroom` would bypass this check, right?

**Pull request merged**